recipes-bsp/secure-boot-otp-provisioning/secure-boot-otp-provisioning_0.1.bb

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) Siemens AG, 2022
+#
+# Authors:
+#  Su Bao Cheng <baocheng.su@siemens.com>
+#
+# SPDX-License-Identifier: MIT
+#
+
+inherit dpkg
+
+DESCRIPTION = "Secure Boot OTP key provisioning tool"
+
+DEBIAN_BUILD_DEPENDS = "openssl, u-boot-tools, device-tree-compiler"
+
+SRC_URI = " \
+    file://its \
+    file://keys \
+    file://make-otpcmd.sh \
+    file://rules"
+
+do_prepare_build[cleandirs] += "${S}/debian"
+do_prepare_build() {
+    deb_debianize
+    cd ${S}
+    cp -rPf ../its ${S}/its
+    ln -sf ../keys ${S}/keys
+    ln -sf ../make-otpcmd.sh ${S}
+    echo "otpcmd.bin /usr/lib/secure-boot-otp-provisioning/" > \
+            ${S}/debian/secure-boot-otp-provisioning.install
+}
+
+OTPCMD_MODE ?= "provision"
+OTP_MPK ?= "./keys/custMpk.pem"
+OTP_SMPK ?= "./keys/custSmpk.pem"
+OTP_BMPK ?= ""
+OTPCMD_KEYS ?= "${OTP_MPK} ${OTP_SMPK} ${OTP_BMPK}"
+
+dpkg_runbuild_prepend() {
+    export OPTCMD_PARAS="${OTPCMD_MODE} ${OTPCMD_KEYS}"
```

Comment:
You know that this pattern will soon be dead in Isar (sbuild...)?

Reply:
Not checked isar yet...

Reply:
More future-proof is a rules template that receives the values of the variables during templating.